src/schemas/Program/index.js

```diff
@@ -352,10 +351,19 @@ const resolvers = {
 				: resolvePublicSingleProgram(shortName);
 		},
 
-		programs: async (obj, args, context, info) => {
+		programs: async (obj, args, context) => {
 			const { egoToken } = context;
-			return resolvePrivateProgramList(egoToken);
+			const { dataCenter } = args;
+
+			const programs = await resolvePrivateProgramList(egoToken);
+
+			const filteredPrograms = programs.filter(
+				(program) => !dataCenter || program.dataCenter?.shortName === dataCenter,
```

Comment:
I think you can avoid the filter iterations completely if there's no dataCenter arg?

Reply:
Yes! Updated